autosklearn/data/feature_validator.py

```diff
@@ -0,0 +1,399 @@
+            raise ValueError("Auto-sklearn only supports Numpy arrays, Pandas DataFrames,"
+                             " scipy sparse and Python Lists, yet, the provided input is"
+                             " of type {}".format(
+                                 type(X)
+                             ))
+
+        if isinstance(X, list):
+            try:
+                np.array(X).astype(np.float64)
+            except ValueError as e:
+                raise ValueError("When providing a list of features to Auto-sklearn, "
+                                 "this list of elements must contain only numerical values "
+                                 "as it will be converted to Numpy. Casting to Numpy failed "
+                                 "with exception: {}".format(e))
+
+        if self.data_type is None:
+            self.data_type = type(X)
+        if self.data_type != type(X):
+            warnings.warn("Auto-sklearn previously received features of type %s "
+                          "yet the current features have type %s. Changing the dtype "
+                          "of inputs to an estimator might cause problems" % (
+                                str(self.data_type),
+                                str(type(X)),
+                             ),
+                          )
+
+        # Do not support category/string numpy data. Only numbers
+        if hasattr(X, "dtype"):
+            if not np.issubdtype(X.dtype.type, np.number):  # type: ignore[union-attr]
+                raise ValueError(
+                    "When providing a numpy array to Auto-sklearn, the only valid "
+                    "dtypes are numerical ones. The provided data type {} is not supported."
+                    "".format(
+                        X.dtype.type,  # type: ignore[union-attr]
+                    )
+                )
+
+        # Then for Pandas, we do not support Nan in categorical columns
+        if hasattr(X, "iloc"):
+            enc_columns, _ = self._get_columns_to_encode(X)
+            if len(enc_columns) > 0:
+                if np.any(pd.isnull(
+                    X[enc_columns].dropna(  # type: ignore[call-overload]
+                        axis='columns', how='all')
+                )):
+                    # Ignore all NaN columns, and if still a NaN
+                    # Error out
+                    raise ValueError("Categorical features in a dataframe cannot contain "
+                                     "missing/NaN values. The OrdinalEncoder used by "
+                                     "Auto-sklearn cannot handle this yet (due to a "
+                                     "limitation on scikit-learn being addressed via: "
+                                     "https://github.com/scikit-learn/scikit-learn/issues/17123)"
+                                     )
+            if self.enc_columns and set(self.enc_columns) != set(enc_columns):
```

Comment:
I don't think doing a set comparison here is a good idea. If the order changes, this will probably mess up the future pipeline?

Reply:
I have removed the set()